pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -334,6 +334,94 @@ var _ = Describe("Snapshotter", func() {
+			store, err = snapstore.GetSnapstore(snapstoreConfig)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		Context("Previous full snapshot was taken more than 24hrs before, so FullSnapshot missed", func() {
+			It("should return true", func() {
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 2 days before
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-2, currentHour, currentMin, 0, 0, time.Local),
+				}
+				isFullSnapMissed := ssr.IsScheduledFullSnapshotMissed()
+				Expect(isFullSnapMissed).Should(BeTrue())
+			})
+		})
+
+		Context("Previous full snapshot was taken exactly at scheduled snapshot time, no FullSnapshot missed", func() {
+			It("should return false", func() {
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 1 days before at exactly at scheduled time
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-1, (currentHour+2)%24, (currentMin+1)%60, 0, 0, time.Local),
+				}
+				isFullSnapMissed := ssr.IsScheduledFullSnapshotMissed()
+				Expect(isFullSnapMissed).Should(BeFalse())
+			})
+		})
+
+		Context("Previous snapshot was taken within 24hrs and not gonna missed schedule fullsnapshot", func() {
+			It("should return false", func() {
+				scheduleHour := (currentHour + 4) % 24
+				snapshotterConfig := &brtypes.SnapshotterConfig{
+					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", currentMin, scheduleHour),
+				}
+
+				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Previous full snapshot was taken 4hrs 10 mins before startup
+				ssr.PrevFullSnapshot = &brtypes.Snapshot{
+					CreatedOn: time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), (currentHour-4)%24, (currentMin-10)%60, 0, 0, time.Local),
+				}
+				isFullSnapMissed := ssr.IsScheduledFullSnapshotMissed()
+				Expect(isFullSnapMissed).Should(BeFalse())
+			})
+		})
+
+		Context("Previous snapshot was taken within 24hrs and gonna miss 24hrs of schedule fullsnapshot window", func() {
+			It("should return true", func() {
```

Comment:
you should also add a case to test if there was no previous snapshot taken

Reply:
added. PTAL